dns.tf

```diff
@@ -1,9 +1,23 @@
-data "exoscale_domain" "this" {
-  name  = local.base_domain
+resource "exoscale_domain" "this" {
```

Comment:
If a domain is used, I don't think this module should create it, because you most likely won't want to delete it if the stack is destroyed. And it should be in line with the behavior of other cluster modules in that regard.